WordPress/gutenberg
# Fix post featured image scaling on the front

## What?

In the block editor, the post featured image `<figure>` wrapper has a set height. This height was missing on the front, so the image did not look the same in the editor and front.
This pr adds the height to the front, both to the `<figure>` and the `<a>` which is used when the image is linked.

Closes [#42322](https://github.com/WordPress/gutenberg/issues/42322)
## Why?

The image did not look the same in the editor and front when image height and scaling was set in the block settings.

## How?

Updates index.php and adds the height to the `get_block_wrapper_attributes` and the `<a>` when applicable.

## Testing Instructions
When testing this we need to take extra care to make sure that the border and border radius settings still work.

1. Make sure that your WordPress install has a few recent posts with a lot of text content and a featured image. - This is needed because we are going to use a query loop and use the long content to stretch the image.
2. Create a new post.
3. Copy the example code from the issue into the post:
```

<div class="wp-block-query">

<div class="wp-block-columns has-cyan-bluish-gray-background-color has-background">
<div class="wp-block-column" style="flex-basis:50%">

</div>



<div class="wp-block-column" style="flex-basis:50%"></div>
</div>

</div>

```
3. The featured image should have a 100% height and "scale" set to cover, and stretch the height of the content.
4. Save and view the front. Confirm that the image looks the same on the front.

5. Continue to test the remaining settings. Confirm that the image looks correct on the front for each step.
For example:
- Enable and disable the linked image setting
- Set the image width to 45px
- Change the scale to contain and fill (The scale settings only show if width or height are adjusted).
- Change the border styles including radius
- Add an overlay

### Testing Instructions for Keyboard


## Screenshots or screencast
